### PR TITLE
Add Tangram namespace

### PR DESCRIPTION
**Pull request by blair1618**
All headers and source files are now within the Tangram namespace. 

I don't like touching many files at once, but it had to be done eventually and it was becoming problematic for the refactoring work I'm doing. 

There are several formatting conventions to choose from regarding namespaces. I've opted to not add indentation for namespaces, a convention followed by LLVM, Valhalla, and others (furthermore, adding indentation for a namespace would require changing every existing line of code in the repo for this pr :P )

### CODE REVIEW

File: core/src/data/dataSource.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include "tile/tileManager.h"
 #include "labels/labels.h"
 
-//---- DataSource Implementation----
+namespace Tangram {
```
Comment (tallytalwar) [inline core/src/data/dataSource.cpp]: I would have done "using namespace Tangram" in a cpp file.

File: core/src/gl/texture.cpp [hunk 2]
```diff
@@ -209,3 +211,5 @@ void Texture::invalidateAllTextures() {
     ++s_validGeneration;
 
 }
+
+}
```
Comment (tallytalwar) [inline core/src/gl/texture.cpp]: github's :heart: for newlines.

File: core/src/data/dataSource.cpp (hunk 1 above)
Comment (blair1618) [inline core/src/data/dataSource.cpp]: That's a possibility. It's functionally equivalent, right?

Comment (tallytalwar) [inline core/src/data/dataSource.cpp]: Yes its functionally equivalent.

**Comment by blair1618** [comment]
An minor drawback of this formatting is that Xcode seems totally unable to conceive the idea that we might not want to indent within namespaces, I can't even find an option to change it's behavior.

**Comment by tallytalwar** [comment]
I use xcode for debugging purpose :P... 
I can have my VI configured to handle this (and probably my xvim for xcode) :D.

**Comment by blair1618** [comment]
Another possibility here is to indent namespaces in headers (to please Varun and Xcode) and then use `using namespace Tangram;` in implementation files.

**Comment by tallytalwar** [comment]
Ohh please me :D. I said I am in love with namespace indentation but I am already married :P.

File: core/src/data/dataSource.cpp (hunk 1 above)
Comment (blair1618) [inline core/src/data/dataSource.cpp]: Actually I've just realized that these are *not* totally equivalent. With 'using', anything not in a class must have its name prefixed by the relevant namespace.

**Comment by hjanetzek** [comment]
maybe a script similar to the uncrustify suggestion helps for xcode http://stackoverflow.com/questions/1035366/is-it-possible-to-customize-the-indent-style-of-xcode

**Comment by blair1618** [comment]
Thanks @hjanetzek I'll check out that script, I'm also going to try the beta for the next xcode, just in case Apple fixed this already (unlikely).

File: core/src/data/dataSource.cpp (hunk 1 above)
Comment (tallytalwar) [inline core/src/data/dataSource.cpp]: Also I was reading more on this ... Its recommended not to use "using" unless really needed and that too in limited scopes. So I am good with this.

**Pull request merged**